x-pack/legacy/plugins/ml/server/models/job_service/new_job/categorization/validation_results.ts

```diff
@@ -0,0 +1,160 @@
+} from '../../../../../common/constants/new_job';
+import {
+  FieldExampleCheck,
+  CategoryFieldExample,
+  VALIDATION_RESULT,
+} from '../../../../../common/types/categories';
+import { getMedianStringLength } from '../../../../../common/util/string_utils';
+
+const VALID_TOKEN_COUNT = 3;
+const MEDIAN_LINE_LENGTH_LIMIT = 400;
+const NULL_COUNT_PERCENT_LIMIT = 0.75;
+
+export class ValidationResults {
+  private _results: FieldExampleCheck[] = [];
+
+  public get results() {
+    return this._results;
+  }
+
+  public get overallResult() {
+    if (this._results.some(c => c.valid === CATEGORY_EXAMPLES_VALID_STATUS.INVALID)) {
+      return CATEGORY_EXAMPLES_VALID_STATUS.INVALID;
+    }
+    if (this._results.some(c => c.valid === CATEGORY_EXAMPLES_VALID_STATUS.PARTIALLY_VALID)) {
+      return CATEGORY_EXAMPLES_VALID_STATUS.PARTIALLY_VALID;
+    }
+    return CATEGORY_EXAMPLES_VALID_STATUS.VALID;
+  }
+
+  private _resultExists(id: VALIDATION_RESULT) {
+    return this._results.some(r => r.id === id);
+  }
+
+  public createTokenCountResult(sortedExamples: CategoryFieldExample[], sampleSize: number) {
+    const validExamplesSize = sortedExamples.filter(e => e.tokens.length >= VALID_TOKEN_COUNT)
+      .length;
+    const percentValid = sampleSize === 0 ? 0 : validExamplesSize / sampleSize;
+
+    let valid = CATEGORY_EXAMPLES_VALID_STATUS.VALID;
+    if (percentValid < CATEGORY_EXAMPLES_ERROR_LIMIT) {
+      valid = CATEGORY_EXAMPLES_VALID_STATUS.INVALID;
+    } else if (percentValid < CATEGORY_EXAMPLES_WARNING_LIMIT) {
+      valid = CATEGORY_EXAMPLES_VALID_STATUS.PARTIALLY_VALID;
+    }
+
+    const message = i18n.translate(
+      'xpack.ml.models.jobService.categorization.messages.tokenLengthValidation',
+      {
+        defaultMessage:
+          '{number} field {number, plural, zero {value} one {value} other {values}} analyzed, {percentage}% contain {validTokenCount} or more tokens.',
+        values: {
+          number: sampleSize,
+          percentage: Math.floor(percentValid * 100),
+          validTokenCount: VALID_TOKEN_COUNT,
+        },
+      }
+    );
+
+    if (
+      this._resultExists(VALIDATION_RESULT.TOO_MANY_TOKENS) === false &&
+      this._resultExists(VALIDATION_RESULT.FAILED_TO_TOKENIZE) === false
+    ) {
+      this._results.unshift({
+        id: VALIDATION_RESULT.TOKEN_COUNT,
+        valid,
+        message,
+      });
+    }
+  }
+
+  public createMedianMessageLengthResult(examples: string[]) {
+    const median = getMedianStringLength(examples);
+
+    if (median > MEDIAN_LINE_LENGTH_LIMIT) {
+      this._results.push({
+        id: VALIDATION_RESULT.MEDIAN_LINE_LENGTH,
+        valid: CATEGORY_EXAMPLES_VALID_STATUS.PARTIALLY_VALID,
+        message: i18n.translate(
+          'xpack.ml.models.jobService.categorization.messages.medianLineLength',
+          {
+            defaultMessage:
+              'The median length for the field values analysed is over {medianLimit} characters.',
```

Comment:
I'm thinking this should be `analyzed`? You use `analyze` elsewhere.

Reply:
fixed in d2e876af534d27a1948cc07ba2c031240ddf5b40